Feat #52 AppBar 제작

dong-fa opened the pull request:
뒤로가기 버튼 history.go(-1) 사용해서 구현
페이지 타이틀은 props로 전달
아이콘 개수별, 텍스트 버튼별 컴포넌트 개별 생성

컴포넌트 이름 재정의 필요한지 의견 필요

사용예시
`return (
      <AppBarNomal title={'페이지 타이틀'}></AppBarNomal>
  ); `
  

  ------------
  
개별로 만들었던 컴포넌트 1개로 통합

title, icon, text, counter Props로 동적 렌더링 가능

불필요한 함수 제거

dong-fa committed:
Feat #52 AppBar 제작

컴포넌트 기본 스타일 지장
컴포넌트 기본 로직 정의

dong-fa committed:
Feat #52 AppBar 컴포넌트 제작

뒤로가기 버튼 history.go(-1) 사용
페이지 타이틀은 props로 전달
아이콘 개수별, 텍스트 버튼별 컴포넌트 개별 생성

jennie-0530 commented on src/components/appbar/AppBarCount.tsx:
```diff
@@ -0,0 +1,75 @@
+
+export default AppBarCount;
+
+const AppBarContainer = styled.div`
```
styled component 변수들은 하단이 아니라 상단에 모아두는 것이 어떨까요?

dong-fa replied:
그러고보니 저만 아래에 작성하고 있었네요 수정 하겠읍니다. :)

jennie-0530 commented on src/components/appbar/AppBarCount.tsx:
```diff
@@ -0,0 +1,75 @@
+  gap: 10px;
+`;
+
+const GobackButton = styled.button`
```
카멜 케이스를 사용해서 GoBackButton으로 수정하는 것은 어떠실까요?!

sungin46 commented on src/components/appbar/AppBarCount.tsx:
```diff
@@ -0,0 +1,75 @@
+
+export default AppBarCount;
+
+const AppBarContainer = styled.div`
+  border-bottom: 1px solid ${color.gray200};
+  width: 1024px;
+  height: 48px;
+  padding: 12px 24px 12px 16px;
+  display: flex;
+  justify-content: center;
+  align-items: flex-start;
+  gap: 10px;
+`;
```
반복되는 컴포넌트들이 많은데 한번에 묶어서 쓰면 좋을거같아요 🙏

dong-fa committed:
Fix #52 PR 1차 피드백 수정

스타일 코드 상단으로 이동

GobackButton => GoBackButton 변경

0SCAR0421 commented on src/components/appbar/AppBarCount.tsx:
```diff
@@ -0,0 +1,75 @@
+    <AppBarContainer>
+      <GoBackButton
+        onClick={() => {
+          history.go(-1);
```
history 객체의 함수는 실행시 page refresh가 발생하지 않나요?

dong-fa replied:
history.pushstate()를 쓰면 리프레시 없이 페이지 이동을 할 수 있는걸로 알고 있습니다. 
이거 이용해서 페이지 이동하는걸 찾아봐야겠네요.
추가로 혹시나 있을 브라우저 뒤로가기 버튼을 막는것도 고려해봐야 될 것 같기도 하네요.

0SCAR0421 commented on src/components/appbar/AppBarNomal.tsx:
Nomal -> Normal 
오타인거 같습니다

dong-fa replied:
확인

0SCAR0421 commented on src/components/appbar/AppBarNomal.tsx:
```diff
@@ -0,0 +1,47 @@
+  flex: 1;
+`;
+
+const AppBarNomal = ({ title }: IappBarProps): JSX.Element => {
```
여기도 오타인거 같습니다

dong-fa replied:
확인

0SCAR0421 commented on src/components/appbar/AppBarTextOne.tsx:
```diff
@@ -0,0 +1,72 @@
+`;
+
+const AppBarTextOne = ({ title }: IappBarProps): JSX.Element => {
+  const onClickAction1 = () => {
+    null;
+  };
```
정확히 무슨 동작을 하는 함수인지 의도가 불명확한거 같습니다

0SCAR0421 commented on src/components/appbar/AppBarTextOne.tsx:
```diff
@@ -0,0 +1,72 @@
+        <Body1>{title}</Body1>
+      </PageNameWarp>
+      <IconWarp>
+        <ActionBtn1 onClick={onClickAction1}>
+          <Body1>버튼</Body1>
+        </ActionBtn1>
```
이 버튼을 클릭했을 때 onClickAction1이 실행되는 건 알겠는데 위에서도 언급했듯이 사용자가 원하는 동작은 일어나지 못하는거 같습니다 확인해주세요

0SCAR0421 reviewed (changes requested):
- 모든 컴포넌트의 onClickAction 함수의 의도가 불명확함. 
사용자가 원하는 함수를 동작시키기 위해서는 추가적인 작업이 필요할것으로 보임.
- history.go 함수는 페이지 새로고침이 발생하는 것이 아닌지 확인 필요.
- 단순 오타 수정

dong-fa committed:
Feat #54 AppBar 로직 수정

개별로 만들었던 컴포넌트 1개로 통합

title, icon, text, counter Props로 동적 렌더링 가능

불필요한 함수 제거

dong-fa committed:
Feat #54 AppBar 로직 수정

icon, text Props 배열 형태로 변경

예시
icon {iconName , func }[]
text {textName , func}[]

각 버튼에 맞는 타입과 실행 함수를 넘겨서 동적으로 렌더링 가능

0SCAR0421 commented on src/components/appbar/index.ts:
```diff
@@ -0,0 +1 @@
+export * from './AppBar';
```
export {default as AppBar} from './AppBar' 형식이어야 하는거 같습니다

dong-fa replied:
확인

0SCAR0421 commented on src/components/appbar/AppBar.tsx:
```diff
@@ -0,0 +1,121 @@
+    <AppBarContainer>
+      <GoBackButton
+        onClick={() => {
+          history.go(-1);
```
리프레쉬 이슈 반영안된거 같습니다 확인해주세요

dong-fa replied:
페이지 이동이 어떻게 되었느냐에 따라 리프레시가 작동 여부 확인 되었습니다.
일반 html 페이지 이동처럼 새로운 페이지를 이동한다면 popstate이벤트가 발생하게 되면서  history.go(-1)작동시 리프레시가 발생하는걸로 확인되었습니다.
하지만 popstate이벤트 없이 라우트 이동을 한다면 history.go(-1)가 작동해도 리프레시 없이 작동하는걸 확인했습다.

0SCAR0421 commented on src/components/appbar/AppBar.tsx:
```diff
@@ -0,0 +1,121 @@
+        <IconWarp>
+          {icon.map((iconItem, index) => (
+            <IconBtn key={index} onClick={iconItem.func}>
+              <Graphic name={`${iconItem.iconName}`} />
```
상당수의 아이콘이 AppBar의 영역보다 크게 나옵니다 확인해주세요

dong-fa replied:
확인 피그마 다시 확인하고 수정하겠습니다.

sungin46 reviewed (commented):
죠습니다 👍

jennie-0530 commented on src/components/appbar/AppBar.tsx:
```diff
@@ -0,0 +1,121 @@
+  gap: 20px;
+`;
+
+const IconBtn = styled.button`
```
저희 변수명은 줄이지 않고 쓰기로 했어요! IconButton으로 수정 부탁드려요~! 다른 Btn도 수정 부탁드립니다!

dong-fa replied:
확인

dong-fa committed:
Feat #54 PR 피드백 반영

history.go(-1) 유지, popstate에 따라 리프레시 작동 달라짐

index 수정

icon 변경

컴포넌트 이름 수정

size 별 렌더링 수정, 타입 추가

jennie-0530 commented:
고생하셨읍니다

sungin46 commented:
is 빠킹 Good 👍

dong-fa committed:
fix #52 실행 함수 추가

full일때 아이콘과 전달 받을 실행 함수 누락 확인, 추가함

sungin46 reviewed (commented):
is 바뀐 Good 👍
